secure_redis/decorators.py

```diff
@@ -0,0 +1,159 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from functools import wraps
+from collections import defaultdict
+import logging
+import datetime
+
+from django.conf import settings as global_settings
+import rq.utils
+from rq.compat import string_types
+import django_rq
+from django_rq.queues import get_queue
+from rq.defaults import DEFAULT_RESULT_TTL
+from rq.queue import Queue
+from . import settings
+
+logger = logging.getLogger(__name__)
+
+
+_serializer = None
+
+
+def get_serializer():
+    global _serializer
+    if not _serializer:
+        from . import serializer
+        _serializer = serializer.SecureSerializer(settings.get_secure_cache_opts())
+    return _serializer
+
+
+def execute(method_name, *args, **kwargs):
+    return rq.utils.import_attribute(method_name)(*args, **kwargs)
+
+
+def secure_job_proxy(*args, **kwargs):
+    args2 = list(args)
+    f_name = get_serializer().loads(args2.pop(0))
+    args = list(get_serializer().loads(args2.pop(0)))
+    kwargs = get_serializer().loads(args2.pop())
+    return execute(f_name, *args, **kwargs)
+
+
+def rq_job(func_or_queue, connection=None, *args, **kwargs):
+    class _rq_job(object):
+        def __init__(self, queue, connection=None, timeout=None,
+                     result_ttl=DEFAULT_RESULT_TTL, ttl=None):
+            """A decorator that adds a ``delay`` method to the decorated function,
```

Comment:
@GhalebKhaled I think you will need to update this docstring to reflect the current implementation of this decorator. Please describe the methods you added below other than `delay`